docs/core/deploying/native-aot/diagnostics.md

```diff
@@ -1,70 +1,78 @@
 
-The native AOT compiler generates information about line numbers, types, locals, and parameters. The native debugger allows inspecting stack trace and variables, stepping into or over source lines, or setting line breakpoints.
+After publishing, Native AOT applications are true native binaries. The managed debugger will not work on them. However, the Native AOT compiler generates fully native executable files that can be debugged by native debuggers on your platform of choice (e.g. WinDbg or Visual Studio on Windows, and gdb or lldb on Unix-like systems).
+
+The NativeAOT compiler generates information about line numbers, types, locals and parameters. The native debugger will let you inspect stack trace and variables, step into/over source lines, or set line breakpoints.
+
+To debug managed exceptions, set a breakpoint on the `RhThrowEx` method -- this method is called whenever a managed exception is thrown.
 
 Collecting a [dump](../../diagnostics/dumps.md) file for a native AOT application involves some manual steps in .NET 8.
 
-### Heap analysis
+### Visual Studio-specific notes
+
+You can launch a NativeAOT-compiled executable under the VS debugger by opening it in the Visual Studio IDE. In the `File`` menu, choose `Open Project/Solution...`` and navigate to the native executable. You can set breakpoints as needed. To start debugging the EXE, choose the `Start Debugging`` option from the Debug menu.
+
+To set a breakpoint that breaks whenever an exception is thrown, choose the Breakpoints option from the `Debug -> Windows` menu. In the new window, select `New -> Function` breakpoint. Specify `RhThrowEx`` as the Function Name and leave the Language option at "All Languages" (do not select C#).
+
+To see what exception was thrown, start debugging (`Debug -> Start Debugging` or `F5`), open the Watch window (`Debug -> Windows -> Watch`) and add following expression as one of the watches: `(S_P_CoreLib_System_Exception*)@rcx`. This leverages the fact that at the time `RhThrowEx` is called, the x64 CPU register RCX contains the thrown exception. You can also paste the expression into the Immediate Window; the syntax is the same as for watches.
+
+### Importance of the symbol file
+
+In native AOT, symbol-file-dependent diagnostics (such as debugging [PerfView](https://github.com/microsoft/perfview) callstacks) don't work at all unless the diagnostic tool has access to the monolithic PDB that was generated when the application was compiled. In .NET, symbol-file-dependent diagnostics generally work just fine, or even great, even if the diagnostic tool has no access to any PDBs that were generated when the application was compiled. That is, symbols for the .NET runtime and the .NET libraries can be pulled from symbol servers, and diagnostic tools can still show function names and accurate call stacks even without access to the compile-time PDBs.
```

Comment:
Let me try to cleanup this paragraph.

Reply:
I think this should also link to the 'Native debug information' section in the overview (and probably say 'symbol file' or 'debug information file' instead of 'monolithic PDB')

Reply:
This is nested inside the "Native debugging information" section. Are you thinking there should be a link to the top of the section?

Reply:
Sorry that was really ambiguous - this is the page/section I was referring to: https://learn.microsoft.com/dotnet/core/deploying/native-aot#native-debug-information